### PR TITLE
Updated mail-host-notification and mail-service-notification

### PR DESCRIPTION
**Pull request by Tqnsls**
I added support for host.notes and service.notes if it is specified on the host

### CODE REVIEW

**Comment by Tqnsls** [comment]
Verified

**Comment by bobapple** [comment]
@cla-bot check

**Comment by Tqnsls** [comment]
> After that, please reply here with a comment and we'll verify.

File: etc/icinga2/scripts/mail-host-notification.sh [hunk 1]
```diff
@@ -63,7 +64,7 @@ urlencode() {
 }
 
 ## Main
-while getopts 4:6::b:c:d:f:hi:l:n:o:r:s:t:v: opt
+while getopts 4:6::b:c:d:f:hi:l:n:o:r:s:t:v:X:x: opt
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-host-notification.sh:67]: What's the lower x for?

Comment (Tqnsls) [reply etc/icinga2/scripts/mail-host-notification.sh]: Corrected, thx

File: etc/icinga2/scripts/mail-service-notification.sh [hunk 2]
```diff
@@ -79,6 +81,8 @@ do
     i) ICINGAWEB2URL=$OPTARG ;;
     l) HOSTNAME=$OPTARG ;; # required
     n) HOSTDISPLAYNAME=$OPTARG ;; # required
+    X) HOSTNOTES="${OPTARG}" ;;
+    x) SERVICENOTES="${OPTARG}" ;;
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-service-notification.sh:85]: It won't hurt as you did, but _consider_ writing the assignment like the others. See also:

https://unix.stackexchange.com/questions/97560/are-quotes-needed-for-local-variable-assignment

File: etc/icinga2/scripts/mail-host-notification.sh [hunk 3]
```diff
@@ -77,6 +78,7 @@ do
     l) HOSTNAME=$OPTARG ;; # required
     n) HOSTDISPLAYNAME=$OPTARG ;; # required
     o) HOSTOUTPUT=$OPTARG ;; # required
+    X) HOSTNOTES="${OPTARG}" ;;
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-host-notification.sh:81]: Same here.

File: etc/icinga2/scripts/mail-host-notification.sh [hunk 4]
```diff
@@ -129,6 +131,12 @@ if [ -n "$HOSTADDRESS6" ] ; then
 IPv6:	 $HOSTADDRESS6"
 fi
 
+## Check whether host_notes was specified.
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-host-notification.sh:134]: I'd write host_notes more human-readable -like IPv4 above- i.e. `host notes`.

File: etc/icinga2/scripts/mail-service-notification.sh [hunk 5]
```diff
@@ -134,6 +138,18 @@ if [ -n "$HOSTADDRESS6" ] ; then
 IPv6:    $HOSTADDRESS6"
 fi
 
+## Check whether host_notes was specified.
+if [ -n "${HOSTNOTES}" ] ; then
+  NOTIFICATION_MESSAGE="${NOTIFICATION_MESSAGE}
+Host-Notes:    ${HOSTNOTES}"
+fi
+
+## Check whether service_notes was specified.
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-service-notification.sh:147]: Same here.

File: etc/icinga2/scripts/mail-service-notification.sh [hunk 6]
```diff
@@ -134,6 +138,18 @@ if [ -n "$HOSTADDRESS6" ] ; then
...
+if [ -n "${SERVICENOTES}" ] ; then
+  NOTIFICATION_MESSAGE="${NOTIFICATION_MESSAGE}
+Service-Notes:   ${SERVICENOTES}"
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-service-notification.sh:150]: Same here.

File: etc/icinga2/scripts/mail-host-notification.sh [hunk 7]
```diff
@@ -129,6 +131,12 @@ if [ -n "$HOSTADDRESS6" ] ; then
...
+if [ -n "${HOSTNOTES}" ] ; then
+        NOTIFICATION_MESSAGE="${NOTIFICATION_MESSAGE}
+Host-Notes:    ${HOSTNOTES}"
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-host-notification.sh:137]: And here.

File: etc/icinga2/scripts/mail-host-notification.sh [hunk 8]
```diff
@@ -129,6 +131,12 @@ if [ -n "$HOSTADDRESS6" ] ; then
 IPv6:	 $HOSTADDRESS6"
 fi
 
+## Check whether host_notes was specified.
+if [ -n "${HOSTNOTES}" ] ; then
```
Comment (Al2Klimov) [inline etc/icinga2/scripts/mail-host-notification.sh:135]: And while you're on it: the { } are not necessary here.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
LGTM in general. Just a few cosmetic stuff:

File: etc/icinga2/scripts/mail-service-notification.sh (hunk 2 above)
Comment (Tqnsls) [reply etc/icinga2/scripts/mail-service-notification.sh]: Sure, no problem. 
I always use quotes and curly brackets in bash, as for me it is no disadvantage but rather a security advantage (I know, not necessary in most cases, but it doesn't hurt imho :))
Ref, which I always like to read: https://unix.stackexchange.com/questions/171346/#171347

I actually wondered why the notification.sh-Scripts are not written with  brackets and stuff.

File: etc/icinga2/scripts/mail-host-notification.sh (hunk 4 above)
Comment (Tqnsls) [reply etc/icinga2/scripts/mail-host-notification.sh:134]: Also no problem, I wanted to submit the variables as specific as possible, so host_notes or host.notes was best imho.

**Review by Al2Klimov — DISMISSED** [review]

**Comment by julianbrost** [comment]
Shouldn't this also be included in the corresponding `NotificationCommand`  in `etc/icinga2/conf.d/commands.conf`?

**Comment by Tqnsls** [comment]
> Shouldn't this also be included in the corresponding `NotificationCommand` in `etc/icinga2/conf.d/commands.conf`?

@julianbrost updated, feel free to correct, if something is missing there.

File: etc/icinga2/conf.d/commands.conf [hunk 9]
```diff
@@ -43,6 +43,9 @@ object NotificationCommand "mail-host-notification" {
       required = true
       value = "$notification_hoststate$"
     }
+    "-X" = {
+      required = false
+      value = "$notification_hostnotes$"
     "-t" = {
```
Comment (julianbrost) [inline etc/icinga2/conf.d/commands.conf:49]: A `}` is missing here. Also, you don't even have to put the `required = false` here for optional arguments, see the first few ones in this block for example.

**Review by Al2Klimov — APPROVED** [review]
```
$ cat 9713.conf
include <itl>
include "/Users/aklimov/NET/WS/icinga2/prefix/etc/icinga2/conf.d/commands.conf"

object CheckerComponent "cc" { }
object NotificationComponent "nc" { }

object User "u" {
	email = "root@localhost"
}

object Host "h" {
	check_command = "dummy"
	vars.dummy_state = 2
	notes = "h.n"
}

object Host "s" {
	check_command = "dummy"
	vars.dummy_state = 0
	notes = "s.n"
}

object Service "s" {
	host_name = "s"
	check_command = "dummy"
	vars.dummy_state = 2
	notes = "h.s.n"
}

object Notification "h" {
	host_name = "h"
	command = "mail-host-notification"
	users = [ "u" ]
}

object Notification "he" {
	host_name = "h"
	command = "mail-host-notification-env"
	users = [ "u" ]
}

object Notification "s" {
	host_name = "s"
	service_name = "s"
	command = "mail-service-notification"
	users = [ "u" ]
}

object Notification "se" {
	host_name = "s"
	service_name = "s"
	command = "mail-service-notification-env"
	users = [ "u" ]
}
$ prefix/sbin/icinga2 daemon -x warning -c 9713.conf
[2023-05-08 15:27:55 +0200] warning/PluginNotificationTask: Notification command for object 's!s' (PID: 20764, arguments: '/Users/aklimov/NET/WS/icinga2/prefix/etc/icinga2/scripts/mail-service-notification.sh') terminated with exit code 111, output: ***** Service Monitoring on AlexandersMBP2.int.netways.de *****

s on s is CRITICAL!

Info:    Check was successful.

When:    2023-05-08 15:27:55 +0200
Service: s
Host:    s
Host notes: s.n
Service notes: h.s.n
[2023-05-08 15:27:55 +0200] warning/PluginNotificationTask: Notification command for object 's!s' (PID: 20763, arguments: '/Users/aklimov/NET/WS/icinga2/prefix/etc/icinga2/scripts/mail-service-notification.sh' '-4' '' '-6' '' '-X' 's.n' '-b' '' '-c' '' '-d' '2023-05-08 15:27:55 +0200' '-e' 's' '-l' 's' '-n' 's' '-o' 'Check was successful.' '-r' 'root@localhost' '-s' 'CRITICAL' '-t' 'PROBLEM' '-u' 's' '-x' 'h.s.n') terminated with exit code 111, output: ***** Service Monitoring on AlexandersMBP2.int.netways.de *****

s on s is CRITICAL!

Info:    Check was successful.

When:    2023-05-08 15:27:55 +0200
Service: s
Host:    s
Host notes: s.n
Service notes: h.s.n
[2023-05-08 15:28:40 +0200] warning/PluginNotificationTask: Notification command for object 'h' (PID: 20796, arguments: '/Users/aklimov/NET/WS/icinga2/prefix/etc/icinga2/scripts/mail-host-notification.sh') terminated with exit code 111, output: ***** Host Monitoring on AlexandersMBP2.int.netways.de *****

h is DOWN!

Info:    Check was successful.

When:    2023-05-08 15:28:40 +0200
Host:    h
Host notes: h.n
[2023-05-08 15:28:40 +0200] warning/PluginNotificationTask: Notification command for object 'h' (PID: 20797, arguments: '/Users/aklimov/NET/WS/icinga2/prefix/etc/icinga2/scripts/mail-host-notification.sh' '-4' '' '-6' '' '-X' 'h.n' '-b' '' '-c' '' '-d' '2023-05-08 15:28:40 +0200' '-l' 'h' '-n' 'h' '-o' 'Check was successful.' '-r' 'root@localhost' '-s' 'DOWN' '-t' 'PROBLEM') terminated with exit code 111, output: ***** Host Monitoring on AlexandersMBP2.int.netways.de *****

h is DOWN!

Info:    Check was successful.

When:    2023-05-08 15:28:40 +0200
Host:    h
Host notes: h.n
```

**Pull request merged**